node/lib/util/open.js

```diff
@@ -254,12 +256,20 @@ Opener.prototype.clearAbsorbedCache = function (subName) {
  * @param {String} subName
  * @return {Boolean}
  */
-Opener.prototype.isOpen = co.wrap(function *(subName) {
-    if (!this.d_initialized) {
-        yield this._initialize();
+Opener.prototype.isOpen = function (subName) {
+    if (this.d_initialized) {
+        return this.d_openSubs.has(subName) || (subName in this.d_cachedSubs);
+    } else {
+        const modulesDir = path.join(this.d_repo.path(), "modules");
+        const submodulePath = path.join(modulesDir, subName);
```

Comment:
does  .git always reference HEAD of half opened modules?

Reply:
This method is asking whether a submodule is fully-open not half-open. Half-open submodules are prevented by the check a few lines down.